expression/builtin_convert_charset.go

```diff
@@ -45,6 +64,8 @@ func (c *tidbConvertCharsetFunctionClass) getFunction(ctx sessionctx.Context, ar
 		if err != nil {
 			return nil, err
 		}
+		bf.tp = args[0].GetType().Clone()
+		bf.tp.Charset, bf.tp.Collate = charset.CharsetBin, charset.CollationBin
```

Comment:
It seems that we don't use `bf.tp`, we use `b.args[0].GetType` directly: https://github.com/pingcap/tidb/pull/29905/files#diff-b7c01f9c9ab46f762c72934562d5e01b7ec3750c1b6b6601147b5b2808d30d5cR92.

Reply:
yes, it is not used for `evalString`, it is set the function's return type

Reply:
Got it.